apps/happychat/src/app.jsx

```diff
@@ -79,6 +79,12 @@ async function AppBoot() {
 		}
 	}
 
+	if ( auth ) {
+		window.configData.env_id =
+			auth.url === 'https://happychat.io/customer' ? 'production' : 'development';
+		window.configData.happychat_url = auth.url;
+	}
```

Comment:
We don't need this anymore 🕺🏼  We should delete all `happychat_url` instances from all the code and use the one from session. And to fix the widgets app and change main Calypso, you just need to change this [function](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/happychat/utils.js) the same way you changed this [one](https://github.com/Automattic/wp-calypso/blob/trunk/packages/happychat-connection/src/use-happychat-auth.ts#L14). 

But please create a `url` variable with a JSDoc above it. Because this is a confusing change for someone 1) used to happychat_url config 2) unaware of the context.

Something like this:
```js
	/**
	 * The Happychat endpoint URL is determined by the the /happychat/session endpoint.
	 * The /happychat/session endpoint returns the right env (staging vs production) based on whether the user is proxied.
	 */
	const url: string = ....
```

@klimeryk This is a tiny but important change and might confuse people, is there somewhere we should share it?

He're the reasoning behind it: p1663771182341809-slack-C02T4NVL4JJ